### PR TITLE
px4_fmu-v5 board spi cleanup

### PR DESCRIPTION
**Pull request by dagar**
- px4_fmu-v5 spi handler convert to c++ and use range based for loops instead of first/last defines
 - remove px4 headers
     - use syslog instead of PX4_ERR, PX4_WARN, etc
 - remove dummy BMA, BMI SPIDEVs
 - remove stm32_spi_bus_initialize
     - handled by the actual drivers
 - delete unused static variables

### CODE REVIEW

File: boards/px4/fmu-v5/src/board_config.h [hunk 1]
```diff
@@ -108,7 +108,7 @@
  */
 
 #define PX4_SPI_BUS_SENSORS   1
-#define PX4_SPI_BUS_MEMORY    2
+#define PX4_SPI_BUS_RAMTRON   2
```
Comment (davids5) [inline boards/px4/fmu-v5/src/board_config.h]: MEMORY was the Nuttx naming and matches the DEV ID. Hence the alias.

File: boards/px4/fmu-v5/src/can.c [hunk 2]
```diff
@@ -1,129 +0,0 @@
-/****************************************************************************
```
Comment (davids5) [inline boards/px4/fmu-v5/src/can.c]: I have had request to use the nuttx can driver. It is here to support those uses.

File: boards/px4/fmu-v5/src/spi.cpp [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+	// SPI select is active low, so write !selected to select the device
+	stm32_gpiowrite(spi6selects_gpio[PX4_SPI_DEV_ID(devid)], !selected);
+}
+
+__EXPORT uint8_t stm32_spi6status(FAR struct spi_dev_s *dev, uint32_t devid)
+{
+	return SPI_STATUS_PRESENT;
+}
+#endif // CONFIG_STM32F7_SPI6
+
```
Comment (davids5) [inline boards/px4/fmu-v5/src/spi.cpp]: Why not normalize the CONFIG_STM32F7_SPI6 and CONFIG_STM32F_SPI6 and 
with PX4_CONFIG_SPI1..N and then move this to board common using the hal gpio assessors. Then there would be one SPI CS driven only from BSP.

**Comment by davids5** [comment]
@dagar your call. I prefer the Board config matches the pinout. Usage may vary.

> Questions
> 
> * PX4_SPIDEV_SPI4_CS2
>   

>   * Not used, remove?

https://docs.google.com/spreadsheets/d/1-n0__BYDedQrc_2NHqBenG1DNepAgnHpSGglke-QQwY/edit#gid=912976165

91 | PF | 11 | R6 | PF11 | SPI4_CS2 | Digital OUPUT: Active Low SPI Chip Select Not used if MS56xx is used on SPI Bus 4 |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

> * PX4_SPIDEV_AUX_MEM
>   
>   * supposed to be on SPI1?

Yes. On HB v5 is is on SPI1

118 | PH | 5 | J4 | PH5 | SPIAUX_CS_MEM | Digital OUPUT: Active Low SPI Chip Select for an EEPROM collocated on Sensor board, used for factory set temperature calibration data |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |  
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

**Comment by dagar** [comment]
Bench tested on a pixhawk 4.

File: boards/px4/fmu-v5/src/spi.cpp (hunk 3 above)
Comment (dagar) [reply boards/px4/fmu-v5/src/spi.cpp]: Once this settles I'll think about moving it to a general (board common) location with micro hal usage.

**Comment by dagar** [comment]
@PX4/testflights could you give this a quick test on any fmu-v5 unit (Pixhawk 4, 4 mini)? Thanks!

**Comment by dannyfpv** [comment]
tested on pixhawk4 v5
https://review.px4.io/plot_app?log=1a037a18-a0e3-4f1c-9365-b9716c7b15a7

**Comment by dagar** [comment]
@simonegu I added a commit to remove PX4_FMUV5_RC00.

File: boards/px4/fmu-v5/src/board_config.h [hunk 4]
```diff
@@ -452,14 +444,8 @@
...
-#if defined(PX4_FMUV5_RC00)
-#  define GPIO_VDD_5V_PERIPH_EN         /* PG4  */ (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_CLEAR|GPIO_PORTG|GPIO_PIN4)
-#  define GPIO_xVDD_5V_PERIPH_EN                   GPIO_VDD_5V_PERIPH_EN
-#endif
-#if defined(PX4_FMUV5_RC01)
-#  define GPIO_nVDD_5V_PERIPH_EN         /* PG4  */ (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTG|GPIO_PIN4)
-#  define GPIO_xVDD_5V_PERIPH_EN                    GPIO_nVDD_5V_PERIPH_EN
-#endif
+#define GPIO_nVDD_5V_PERIPH_EN         /* PG4  */ (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTG|GPIO_PIN4)
+#define GPIO_xVDD_5V_PERIPH_EN                    GPIO_nVDD_5V_PERIPH_EN
```
Comment (davids5) [inline boards/px4/fmu-v5/src/board_config.h]: This goes away

Comment (dagar) [reply boards/px4/fmu-v5/src/board_config.h]: Yes it does.

File: boards/px4/fmu-v5/src/board_config.h [hunk 5]
```diff
@@ -121,6 +115,8 @@
 #define GPIO_SPI1_CS2_ICM20602    /* PF3 */  (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTF|GPIO_PIN3)
 #define GPIO_SPI1_CS3_BMI055_GYRO /* PF4 */  (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTF|GPIO_PIN4)
 #define GPIO_SPI1_CS4_BMI055_ACC  /* PG10 */ (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTG|GPIO_PIN10)
+#define GPIO_SPI1_CS5_AUX_MEM     /* PH5  */ (GPIO_OUTPUT|GPIO_PUSHPULL|GPIO_SPEED_2MHz|GPIO_OUTPUT_SET|GPIO_PORTH|GPIO_PIN5)
```
Comment (simonegu) [inline boards/px4/fmu-v5/src/board_config.h]: I think we should port this pin name change also on the pin definition document (https://docs.google.com/spreadsheets/d/1-n0__BYDedQrc_2NHqBenG1DNepAgnHpSGglke-QQwY/edit?usp=sharing) for more clarity

Comment (davids5) [reply boards/px4/fmu-v5/src/board_config.h]: @simonegu - Agreed - Updated See Legend 2/8/2019 and All Pinouts Tabs

**Pull request merged**